frontend/test/metabase-lib/lib/queries/StructuredQuery.unit.spec.js

```diff
@@ -28,6 +30,36 @@ function makeQueryWithAggregation(agg) {
   return makeQuery({ aggregation: [agg] });
 }
 
+function makeQueryWithoutNumericFields() {
+  const virtualCardId = "card__123";
+
+  const questionDetail = {
+    display: "table",
+    visualization_settings: {},
+    dataset_query: {
+      type: "query",
+      database: SAMPLE_DATABASE.id,
+      query: {
+        "source-table": virtualCardId,
+      },
+    },
+  };
+
+  const metadata = createMetadata(state => {
+    const tableWithWithoutNumericFields = {
+      id: virtualCardId,
+      db_id: SAMPLE_DATABASE.id,
+      fields: [ORDERS.ID.id, ORDERS.CREATED_AT.id],
+    };
+    return state.assocIn(
+      ["entities", "tables", virtualCardId],
+      tableWithWithoutNumericFields,
+    );
+  });
+
+  return new Question(questionDetail, metadata).query();
```

Comment:
the only way I can make it work is to create a metadata that contains a saved question's info.

So when a query has this `"source-table": "card__123"`, it can resolve the original question fields correctly.